ratatui-core/src/layout/layout.rs

```diff
@@ -1,24 +1,28 @@
 use alloc::format;
-use alloc::rc::Rc;
+use alloc::sync::Arc;
 use alloc::vec::Vec;
 use core::cell::RefCell;
 use core::iter;
 use core::num::NonZeroUsize;
-use std::{dbg, thread_local};
+use std::dbg;
 
+#[cfg(not(feature = "tls-layout-cache"))]
+use critical_section::Mutex as CsMutex;
 use hashbrown::HashMap;
 use itertools::Itertools;
 use kasuari::WeightedRelation::{EQ, GE, LE};
 use kasuari::{AddConstraintError, Expression, Solver, Strength, Variable};
 use lru::LruCache;
+#[cfg(not(feature = "tls-layout-cache"))]
+use once_cell::sync::Lazy;
 
 use self::strengths::{
     ALL_SEGMENT_GROW, FILL_GROW, GROW, LENGTH_SIZE_EQ, MAX_SIZE_EQ, MAX_SIZE_LE, MIN_SIZE_EQ,
     MIN_SIZE_GE, PERCENTAGE_SIZE_EQ, RATIO_SIZE_EQ, SPACER_SIZE_EQ, SPACE_GROW,
 };
 use crate::layout::{Constraint, Direction, Flex, Margin, Rect};
 
-type Rects = Rc<[Rect]>;
+type Rects = Arc<[Rect]>;
```

Comment:
This would be a breaking change as although the type isn't public (it's a private alias to simplify the code), the actual type appears in the public interface (`pub fn split(&self, area: Rect) -> Rects`), so if this is where we land, then it needs to be called out in the commit (easiest to put this in the PR body to make a squash merge work well) and in a BREAKING changes doc item. (Happy to defer this until we work out what the actual impl looks like though).